lib/src/rules/prefer_iterable_whereType.dart

```diff
@@ -0,0 +1,86 @@
+// Copyright (c) 2018, the Dart project authors. Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:analyzer/dart/ast/ast.dart';
+import 'package:analyzer/dart/ast/visitor.dart';
+import 'package:linter/src/analyzer.dart';
+import 'package:linter/src/util/dart_type_utilities.dart';
+
+const _desc = r'Prefer to use whereType on iterable.';
+
+const _details = r'''
+
+**PREFER** `iterable.whereType<T>()` over `iterable.where((e) => e is T)`.
+
+**BAD:**
+```
+iterable.where((e) => e is MyClass)
+```
+
+**GOOD:**
+```
+iterable.whereType<MyClass>()
+```
+
+''';
+
+class PreferIterableWhereType extends LintRule {
+  PreferIterableWhereType()
+      : super(
+            name: 'prefer_iterable_whereType',
+            description: _desc,
+            details: _details,
+            group: Group.style);
+
+  @override
+  AstVisitor getVisitor() => new Visitor(this);
+}
+
+class Visitor extends SimpleAstVisitor {
+  Visitor(this.rule);
+
+  final LintRule rule;
+
+  @override
+  visitMethodInvocation(MethodInvocation node) {
+    if (node.methodName.name != 'where') return;
+    if (!DartTypeUtilities.implementsInterface(
+        node.target.bestType, 'Iterable', 'dart.core')) {
+      return;
+    }
+
+    final args = node.argumentList?.arguments;
+    if (args.length != 1) return;
```

Comment:
`args` can be `null` here.

Reply:
oops. Isn't there a lint to show that ? :laughing: